tfe/resource_tfe_sentinel_policy.go

```diff
@@ -0,0 +1,167 @@
+package tfe
+
+import (
+	"fmt"
+	"log"
+
+	tfe "github.com/hashicorp/go-tfe"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceTFESentinelPolicy() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceTFESentinelPolicyCreate,
+		Read:   resourceTFESentinelPolicyRead,
+		Update: resourceTFESentinelPolicyUpdate,
+		Delete: resourceTFESentinelPolicyDelete,
+
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"organization": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"policy": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"enforce_mode": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ValidateFunc: func(v interface{}, k string) ([]string, []error) {
+					switch v.(string) {
+					case "advisory", "hard-mandatory", "soft-mandatory":
+						// These are the two valid options, so we return without errors.
+						return nil, nil
+					default:
+						return nil, []error{fmt.Errorf(
+							"%q must be either 'advisory', 'hard-mandatory' or 'soft-mandatory'", k),
+						}
+					}
+				},
+			},
+		},
+	}
+}
+
+func resourceTFESentinelPolicyCreate(d *schema.ResourceData, meta interface{}) error {
+	tfeClient := meta.(*tfe.Client)
+
+	// Get the name and organization.
+	name := d.Get("name").(string)
+	organization := d.Get("organization").(string)
+
+	// Create a new options struct.
+	options := tfe.PolicyCreateOptions{
+		Name: tfe.String(name),
+		Enforce: []*tfe.EnforcementOptions{
+			&tfe.EnforcementOptions{
+				Path: tfe.String(name + ".sentinel"),
```

Comment:
do we want to add validation to `name` to check that the name doesn't have this suffix?

Reply:
That's a god question. According to the API docs the path should always be the `name` of the policy + `.sentinel`. So if the name would be `Sander.sentinel`, the path should be `Sander.sentinel.sentinel`.

But this is a good one to double check to make sure the doc are also covering this corner case. _Checking..._

Reply:
Seems to work as advertised. But will double check with the TFE team just to be sure.

Reply:
Double checked with the team, and my assumptions are correct. So we don't have to check if the name already end with `.sentinel` here.